pkg/apis/core/validation/validation.go

```diff
@@ -6541,6 +6544,22 @@ func ValidateMaxSkew(fldPath *field.Path, maxSkew int32) *field.Error {
 	return nil
 }
 
+// validateMinDomains tests that the argument is a valid MinDomains.
+func validateMinDomains(fldPath *field.Path, minDomains *int32, action core.UnsatisfiableConstraintAction) field.ErrorList {
+	var allErrs field.ErrorList
+	if minDomains != nil {
+		if *minDomains <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath, minDomains, isNotPositiveErrorMsg))
+		}
+
+		// When MinDomains is different from 1, whenUnsatisfiable must be DoNotSchedule.
+		if action != core.DoNotSchedule && *minDomains != 1 {
+			allErrs = append(allErrs, field.Invalid(fldPath, minDomains, fmt.Sprintf("cannot use MinDomains with whenUnsatisfiable=%s", string(action))))
```

Comment:
use the serialized API name and tell them what whenUnsatisfiable must be:
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath, minDomains, fmt.Sprintf("can only use minDomains if whenUnsatisfiable=%s, not %s", string(core.DoNotSchedule), string(action))))
```